Base/src/main/java/io/deephaven/base/FileUtils.java

```diff
@@ -59,19 +61,23 @@ public static void cleanDirectory(File path) {
     }
 
     public static void deleteRecursively(File file) {
-        if (!file.exists()) {
+        if (!Files.exists(file.toPath(), LinkOption.NOFOLLOW_LINKS)) {
```

Comment:
Do we have any reason to expect that the methods you're substituting don't perform identically to the originals?
Are you certain we shouldn't follow links?

Reply:
Yes. The NOFOLLOW_LINKS is important, otherwise a symlink to a non-existing directory will cause the file to not exist, and thus it is not removed - leaving crap around.

The File.delete() member method does not throw exceptions, it just returns false (which the original code asserted on); meaning you do not have access to the underlying IOException that caused problems.  The new method throws an IOException which we can turn into an UncheckedIOException with the filename, so at least you know why it failed (permission denied, etc).

Reply:
I think this "exists" check makes sense for the top-level caller to match the semantics today, but we might want to restructure the implementation so that the recursive calls don't need to make that check. We might think it's a "good safety check", but there isn't atomicity guarantees with it; exists can succeed now, and fail later.

It's worth noting, the semantics today and with this PR afaict is to recurse into symlinked directories and delete all of those files - is that something we _want_ / _need_? We probably shouldn't change this methods behavior in that regard, but we should probably document it and provide an alternative that we may think of as safer.